### PR TITLE
.github/workflows,.travis.yml: migrate lint and a couple tests to Github Actions

### PR DESCRIPTION
**Pull request by meowsbits**
Travis has a new _Credit_ system that ~~coregeth~~ etclabscore/ is using up (1000 minutes / month) causing Travis to not run anymore until the new month rolls over.

This PR migrates the `make lint` check and a couple simple tests to Github Actions to remove some of the minutes from Travis.

It doesn't touch any Travis jobs that handle deploys (on tags). This is possible in the future, but I just want to get the easy stuff moved over first.

### CODE REVIEW

File: .github/workflows/test-linux.yml [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+        id: go
+        uses: actions/setup-go@v2
+        with:
+          go-version: ^1.15
+
+      - uses: actions/checkout@v2
+        with:
+          submodules: recursive
+
+      - run: make test
```
Comment (ziogaschr) [inline .github/workflows/test-linux.yml]: Don't we need a `make all` here?

Comment (meowsbits) [reply .github/workflows/test-linux.yml]: Thanks. Fixed with 4183ae5fae

**Comment by meowsbits** [comment]
I would like to get the `pull_request` and `push` jobs run exclusively (eg. only run `pull_request` jobs when both hooks are triggered).

**Comment by ziogaschr** [comment]
This will be nice and will save us some minutes as well from CI.

**Review by ziogaschr — APPROVED** [review]

**Comment by ziogaschr** [comment]
I approved it, but let's first see if the tests pass here :) Was very fast to approve

**Comment by meowsbits** [comment]
@ziogaschr Note the go1.17 bump

**Comment by meowsbits** [comment]
I am working on this over here for now, to avoid spamming here.
https://github.com/meowsbits/core-geth/pull/21

**Comment by meowsbits** [comment]
Despite the docs, `ANDROID_HOME` is set in Github Actions for the ubuntu-latest image.

https://docs.github.com/en/actions/learn-github-actions/environment-variables
https://github.com/meowsbits/ci-tests/runs/3689090391?check_suite_focus=true

I don't think it's set in Travis. 

When `ANDROID_HOME` isn't set and the `$HOME/Android/Sdk` path doesn't exist, the test is skipped.

I'm going to set the var to empty to force the test to skip. The directory doesn't exist.

**Comment by meowsbits** [comment]
be advised that I force pushed

**Comment by ziogaschr** [comment]
Great find man. I were looking on the wrong direction to solve this one. Thanks for saving us time.
I am fine skipping and already skipped test for android.

**Pull request merged**